src/main/java/com/fundation/logic/model/SearchVideo.java

```diff
@@ -17,49 +18,75 @@
+/** Implements the model class File and the getter and setter´s methods
+* @author Jesus Menacho
+* @version 1.0 *
+*/
+public class SearchVideo implements ISearch{
 static Criteria criteria;
-
     public SearchVideo(Criteria criteria){
         this.criteria = criteria;
     }
-
     public List searchByPath() {
-        if ((criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
-            return null;
-        }
-        if ((criteria.getPath().equals(""))) {
+        if (criteria.getPath() == null) {
             return null;
         }
         final String FILE_PATH = criteria.getPath();
         List<File> listFileAndDirectory = new ArrayList<File>();
         File file = new File(FILE_PATH);
         File[] allSubFiles = file.listFiles();
         for (File fileExtractor : allSubFiles) {
-            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
-                listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
-            }
-            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getAbsolutePath().contains(criteria.getPath())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+            if (fileExtractor.isHidden() == criteria.getFileHidden())
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
                 }
-            }
-            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getName().contains(criteria.getExtension())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+                else
+                {
+                    if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
                 }
             }
-            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getExtension())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+            else
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
```

Comment:
space before and after the logic operator "!="

Reply:
oks. this was fixed